.github/workflows/test_pip_install.yaml

```
@@ -22,21 +22,22 @@ on:
       
 jobs:
   test_release:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v2
-      - name: Get the tags
-        run: git fetch --all --tags
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          fetch-tags: true
       - name: Install Python's "build" package
-        run: python -m pip install build twine --user
+        run: python -m pip install build twine
```

@ryanmrichard @renefritze The issue with the failing PyPI release test is due to the implicit version of `packaging` (24.0) being pulled in during the test. In pypa/twine#1216 (to save you time, it is summarized [here](https://github.com/pypa/twine/issues/1216#issuecomment-2609745412)), it is noted that `setuptools` generates invalid metadata and upgrading `packaging` to >=24.2 with `twine` >=6.1.0 will solve the issue.

I can't tell why, but the latest version of `packaging` isn't being used here (see [latest run](https://github.com/CMakePP/CMinx/actions/runs/16073562749/job/45433674828#step:3:10) where v24.0 was already found), so the workflow continues to fail. It is probably best to pin `packaging` to >=24.2 here to make sure it gets upgraded and stays upgraded in the future.
```suggestion
        run: python -m pip install build twine packaging>=24.2
```

---

Another note to add: A [fix](https://github.com/pypa/twine/pull/1217) for this has been merged as of April 2, 2025, but Twine's latest release at this time ([6.1.0](https://github.com/pypa/twine/releases/tag/6.1.0)) was released Jan 21, 2025. I don't know what their release model looks like, but I'd hope they would release a 6.1.1 soon to officially fix this.

---

I'm not sure if closing words will work here, but closes #171.